scripts/find-mem-leak.cocci

```diff
@@ -2,8 +2,8 @@
 expression x, y;
 statement S;
 @@
-  x = mbedtls_malloc(...);
-  y = mbedtls_malloc(...);
+  x = mbedtls_calloc(...);
+  y = mbedtls_calloc(...);
   ...
```

Comment:
@hanno-arm: My understanding is that cocci is a sort of scripting language that is executed by `spatch`. This scripts essentially allow you to identify patterns and change the code. So in the case of the `find-mem-leak.cocci` script, we are trying to identify code of the form:
```c
x = mbedtls_calloc();
y = mbedtls_calloc();

if( x == NULL || y == NULL )
{
    return;
}
```
Now consider the case when one allocation succeeds and the other fails. Then either `x == NULL && y != NULL` or `x != NULL && y == NULL` and the `if()` statement will evaluate to true and return. This means that we are either leaking the memory allocating for y in the former case and x in the latter. This cocci script helps us flag some of these memory leaks.